test/test_linalg.py

```diff
@@ -121,6 +121,157 @@ def run_test_case(a, b):
         run_test_case(zero_strided, b)
         run_test_case(a, zero_strided)
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_linalg_lstsq(self, device, dtype):
```

Comment:
We should test directly against np.lstsq as a reference on CPU and on CUDA when the driver's available.